### PR TITLE
Better sync logging

### PR DESCRIPTION
**Pull request by pauljamescleary**
* Cleaning up logging around sync to make it easier to query
* Added a log statement in the main health check to better inspect health check failures

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/engine/ZoneSyncHandler.scala [hunk 1]
```diff
@@ -83,11 +83,12 @@ object ZoneSyncHandler extends DnsConversions with Monitored {
       vinyldnsLoader: (Zone, RecordSetRepository) => VinylDNSZoneViewLoader =
         VinylDNSZoneViewLoader.apply): IO[ZoneChange] =
     monitor("zone.sync") {
-      time(s"zone.sync(${zoneChange.zoneId})") {
+      time(s"zone.sync; zoneName='${zoneChange.zone.name}'") {
```
Comment (rebstar6) [inline modules/api/src/main/scala/vinyldns/api/engine/ZoneSyncHandler.scala]: can you throw the change ID up here somewhere too?

File: modules/api/src/main/scala/vinyldns/api/engine/ZoneSyncHandler.scala [hunk 2]
```diff
@@ -142,7 +145,7 @@ object ZoneSyncHandler extends DnsConversions with Monitored {
     }.attempt
       .map {
         case Left(e: Throwable) =>
-          logger.error(s"Encountered error syncing zone ${zoneChange.zone.name}", e)
+          logger.error(s"Encountered error syncing ; zoneName='${zoneChange.zone.name}'", e)
```
Comment (rebstar6) [inline modules/api/src/main/scala/vinyldns/api/engine/ZoneSyncHandler.scala]: change id in this one too

File: modules/core/src/main/scala/vinyldns/core/health/HealthCheck.scala [hunk 3]
```diff
@@ -17,17 +17,21 @@
...
 import cats.effect.IO
+import org.slf4j.LoggerFactory
 
 object HealthCheck {
 
   type HealthCheck = IO[Either[HealthCheckError, Unit]]
 
   case class HealthCheckError(message: String) extends Throwable(message)
 
+  private val logger = LoggerFactory.getLogger("HealthCheck")
```
Comment (rebstar6) [inline modules/core/src/main/scala/vinyldns/core/health/HealthCheck.scala]: not for now, but i pulled in https://github.com/lightbend/scala-logging to the syncer and its nicer on logger creation. you just `val logger = Logger("name")` and go on your way

Comment (rebstar6) [reply modules/core/src/main/scala/vinyldns/core/health/HealthCheck.scala]: for now - generally dont we name loggers by full path? `LoggerFactory.getLogger("vinyldns.core.health.HealthCheck")`

Comment (pauljamescleary) [reply modules/core/src/main/scala/vinyldns/core/health/HealthCheck.scala]: We have to somehow differentiate between an `object` and a `class` for logging :/

I have traditionally omitted the full class name for `object`.  I will double check but we should follow that precedent here

Comment (pauljamescleary) [reply modules/core/src/main/scala/vinyldns/core/health/HealthCheck.scala]: Yep, like boot `private val logger = LoggerFactory.getLogger("Boot")`

Comment (rebstar6) [reply modules/core/src/main/scala/vinyldns/core/health/HealthCheck.scala]: ok thats fine then :)

**Pull request merged**